cli/cli_integration_test.go

```diff
@@ -0,0 +1,246 @@
+
+		// Check database connection.
+		sendLine(`\c`)
+		expectLine(`You are not connected to a database.`)
+
+		// Try connecting to an invalid database.
+		sendLine(`\c invalid`)
+		expectLine(`executing meta command: invalid database: invalid`)
+
+		// Connect to a database.
+		sendLine(`\c db1`)
+		expectLineComp(compWithFormat, `You are now connected to database "db1" ({uuid}).`)
+
+		// Show tables for database.
+		sendLine(`SHOW TABLES;`)
+		expectLine(` _id                     | name                    | owner | updated_by | created_at                | updated_at                | keys  | space_used | description `)
+		expectLine(`-------------------------+-------------------------+-------+------------+---------------------------+---------------------------+-------+------------+-------------`)
+		// TODO(tlt): the system tables are completely masked off because their
+		// order is not guaranteed. When order is guaranteed, update these to
+		// include the actual system tables names.
+		expectLineComp(compWithFormat, ` fb_____________________ | fb_____________________ |       |            | {timestamp} | {timestamp} | false |          0 |             `)
+		expectLineComp(compWithFormat, ` fb_____________________ | fb_____________________ |       |            | {timestamp} | {timestamp} | false |          0 |             `)
+		expectLineComp(compWithFormat, ` fb_____________________ | fb_____________________ |       |            | {timestamp} | {timestamp} | false |          0 |             `)
+		expectLineComp(compWithFormat, ` fb_____________________ | fb_____________________ |       |            | {timestamp} | {timestamp} | false |          0 |             `)
+		expectLineComp(compWithFormat, ` fb_____________________ | fb_____________________ |       |            | {timestamp} | {timestamp} | false |          0 |             `)
+		expectLine(``)
+
+		// expectLine(`STOPPER`)
+
+		////////////////////////////////////////////////////////////
+
+		// End with quit to ensure that fbsql closes without error.
+		sendLine(`\q`)
+
+		// Ensure fbsql quits cleanly.
+		select {
+		case <-didQuit:
+		case <-time.After(time.Second):
+			t.Fatalf("expected fbsql to quit")
+		}
+	})
+}
+
+// compare is used to compare fbsql output written to its Stdout with expected
+// lines.
+type comparer struct {
+	t       *testing.T
+	out     chan byte
+	outline chan []byte
+	exp     chan []byte
+}
+
+func newComparer(t *testing.T) *comparer {
+	return &comparer{
+		t:       t,
+		out:     make(chan byte, 1024),
+		outline: make(chan []byte, 128),
+		exp:     make(chan []byte, 1024),
+	}
+}
+
+func (c *comparer) run() {
+	// Read bytes off output, and for every line (designated by a line feed "\n"),
+	// push the line onto the ouline channel.
+	go func() {
+		var line []byte
+		for {
+			b := <-c.out
+			if b == byte('\n') {
+				c.outline <- line
+				line = []byte{}
+				continue
+			}
+			line = append(line, b)
+		}
+	}()
+}
+
+type comparitor string
```

Comment:
```suggestion
type comparator string
```

Reply:
I'm gonna back out that last commit anyway. It was just stubbing out a test framework, but I need to fix some things on it before it's ready.